clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/DIVStats.java

```diff
@@ -12,21 +12,17 @@
  */
 public class DIVStats {
   private final MetricConfig metricConfig = new MetricConfig();
-  private final WritePathLatencySensor producerSourceBrokerLatencySensor;
-  private final WritePathLatencySensor sourceBrokerLeaderConsumerLatencySensor;
-  private final WritePathLatencySensor producerLocalBrokerLatencySensor;
-  private final WritePathLatencySensor localBrokerFollowerConsumerLatencySensor;
-  private final WritePathLatencySensor leaderProducerCompletionLatencySensor;
+
   private final WritePathLatencySensor leaderDIVCompletionLatencySensor;
+  private final WritePathLatencySensor drainerDIVCompletionLatencySensor;
```

Comment:
Instead of adding one more DIV latency sensor, I would suggest deleting the other one.

Reply:
It's okay to keep only the drainer one. For messages that go through producer, we actually run DIV twice for the same message, not sure if we should call that a tech debt or bug. In any case, keeping only the drainer DIV latency should cover all cases.

Reply:
plz see [my comment](https://github.com/linkedin/venice/pull/993#discussion_r1607339060) above to see if it makes sense to you.. I'd like to keep the one in the consumer thread too.

Reply:
SGTM! No concerns.

Reply:
There should be no difference in implementation between the DIV that runs in the drainer and the one in the consumer... so we should not expect them to have significantly different latency (other than due to noise, e.g. a GC pause that strikes at exactly the right moment to get counted during DIV's execution latency...). So if there is any hunch that DIV execution might be slow, then let's take a look at the existing metric we have for the leader consumption DIV... My expectation is that we will reconfirm that DIV is extremely fast and not worth having a metric for (but I could be wrong, of course). In any case, I still don't see the point of adding one more DIV metric.

Reply:
> so we should not expect them to have significantly different latency

That's a valid point. The reason why we would like to have two DIV metrics is we want to eliminate the noise and have high fidelity metrics data since data would be more accurate when recording happens in the same thread.

There are some nuances between these two metrics. Original intention of introducing DIV completion metrics is to measure the latency from the start of processing a record until the DIV validation completes (not only for DIV process alone)

For, `leader completion DIV`, we have some operations that's unrelated to DIV happen first, for example, processing SOP, checking schema id availability and other validation work. `drainer completion DIV` is much simpler process and only DIV validation happens, so they are not identical.